QMLComponents/boundcontrols/boundcontrolbase.cpp

```diff
@@ -117,44 +117,47 @@ std::string BoundControlBase::getName() const
 	return fq(_control->name());
 }
 
-void BoundControlBase::_readTableValue(const Json::Value &value, const std::string& key, bool hasMultipleTerms, Terms& terms, ListModel::RowControlsValues& allControlValues, const Terms& sourceTerms)
+void BoundControlBase::_readTableValue(const Json::Value &value, const std::string& keyValue, const std::string& keyLabel, bool hasMultipleTerms, Terms& terms, ListModel::RowControlsValues& allControlValues, const Terms& sourceTerms)
 {
 	for (const Json::Value& row : value)
 	{
-		Json::Value keyValue = row[key];
-
-		Term term = Term::readTerm(keyValue);
+		Term term(row, keyValue, keyLabel);
 		if (term.size() > 0)
 		{
-			int termInd = sourceTerms.indexOf(term);
+			int termInd = sourceTerms.indexOfValue(term);
 			if (termInd >= 0)
+			{
 				term.setTypes(sourceTerms[termInd].types());
+				term.setLabel(sourceTerms[termInd].label());
+			}
 			terms.add(term);
 
 			QMap<QString, Json::Value> controlMap;
 			for (auto itr = row.begin(); itr != row.end(); ++itr)
 			{
 				const std::string& name = itr.key().asString();
-				if (name != key)
+				if (name != keyValue)
 					controlMap[tq(name)] = *itr;
 			}
 
-			allControlValues[term.asQString()] = controlMap;
+			allControlValues[term.value()] = controlMap;
 		}
 	}
 }
 
-Json::Value BoundControlBase::_getTableValueOption(const Terms& terms, const ListModel::RowControlsValues& componentValuesMap, const std::string& key, bool hasInteraction, bool keyHasVariables)
+Json::Value BoundControlBase::_getTableValueOption(const Terms& terms, const ListModel::RowControlsValues& componentValuesMap, const std::string& keyValue, const std::string& keyLabel, bool hasInteraction, bool keyHasVariables)
```

Comment:
Why would you need to give the label to get the option?
Shouldnt the label be stored in the option and be accessible via the key?

Or is this function also responsible for adding stuff to the table?

Reply:
The label might be used by the analysis as well, as a title or in a header of a Table for example. The label is also translated, so it's better to translate it once in QML, and then pass the translated string to the analysis.

Reply:
This function generates the option for many kinds op control.